File: Valid Palindrome.cpp
class Solution {
public:

    bool processingLetters(char &c)
    {
        if(c>='A'&&c<='Z')
        {
            c = tolower(c);
            return true; 
        }
        else if((c>='a'&&c<='z')||isdigit(c))
        {
            return true;
        }
        else
        {
            return false;
        }
    }
    
    bool isPalindrome(string s) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        int start = 0;
        int end  = s.size()-1;
        bool flag = true;
        while(start<end)
        {
            if(!processingLetters(s[start]))
                start++;
            else if(!processingLetters(s[end]))
                end--;
            else
            {
                if(s[start]!=s[end])
                {
                    flag = false;
                    break;
                }
                else
                {
                    start++;
                    end--;
                }
            }
                
        }
        
        return flag;
    }
};

File: Valid Palindrome.java
public class Solution {
    public boolean isPalindrome(String s) {
        // Start typing your Java solution below
        // DO NOT write main() function
        String regexp = "[^a-zA-Z0-9]";
        String str1 = s.replaceAll(regexp, "");
        String str2 = new StringBuffer(str1).reverse().toString();
        if(str1.equalsIgnoreCase(str2))
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}